interp/src/environment.rs

```diff
@@ -1,154 +1,164 @@
-use super::primitives;
+//! Environment for interpreter.
+
+//use super::{primitives, update};
 use calyx::ir;
+use serde::Serialize;
+use std::collections::BTreeMap;
 use std::collections::HashMap;
 use std::rc::Rc;
 
-/// Stores information for updates.
-#[derive(Clone, Debug)]
-pub struct Update {
-    /// The cell to be updated
-    pub cell: ir::Id,
-    /// The vector of input ports
-    pub inputs: Vec<ir::Id>,
-    /// The vector of output ports
-    pub outputs: Vec<ir::Id>,
-    /// Map of intermediate variables
-    /// (could refer to a port or it could be "new", e.g. in the sqrt)
-    pub vars: HashMap<String, u64>,
-}
+// #[derive(Serialize, Debug)]
+// struct Cycle (HashMap<String, HashMap<String, u64>>);
+#[derive(Serialize, Debug)]
+struct Cycle(BTreeMap<String, BTreeMap<String, BTreeMap<String, u64>>>);
 
 /// The environment to interpret a Calyx program.
 #[derive(Clone, Debug)]
 pub struct Environment {
-    /// A mapping from cell names to the values on their ports.
-    // XXX: Should not be `pub`.
-    map: HashMap<ir::Id, HashMap<ir::Id, u64>>,
+    /// Stores values of context.
+    /// Maps component names to a mapping from the component's cell names to their ports' values.
+    pub map: HashMap<ir::Id, HashMap<ir::Id, HashMap<ir::Id, u64>>>,
```

Comment:
Reminder that the value of memories should also be in here somewhere. Likely in a separate field, or have an enum that either points to a memory array or a hashmap for port values